odoo/tools/serialization.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+import json
+import xmlrpc.client
+
+from datetime import date, datetime
+
+import odoo
+
+from .func import lazy
+from . import ustr
+
+
+class OdooMarshaller(xmlrpc.client.Marshaller):
+
+    """
+    XMLRPC marshaller capable of converting different python objects used by Odoo
+    """
+
+    dispatch = dict(xmlrpc.client.Marshaller.dispatch)
+
+    def _Marshaller__dump(self, value, write):
+        # we cannot create a hook for recordsets like for datetime and date objects, because
+        # a recordset of say 'res.users' is not of the same type as a recordset of 'sale.order'
+        # and the __dump function checks for type() and not for isinstance() to determine if an
+        # object is marshallable.
+        # to circumvent this, we check if the object is an instance of BaseModel, which applies
+        # to all recordsets, and then we convert it to id
+        if type(value) not in OdooMarshaller.dispatch and isinstance(value, odoo.models.BaseModel):
```

Comment:
`OdooMarshaller.dispatch` could just be `self.dispatch`, though at this point the serialization of dates and datetimes might as well be moved here without bothering with `dispatch`.